Build and Push to GCR (#18)
mozilla/jira-bugzilla-integration: README.md

![build badge](https://github.com/mozilla/jira-bugzilla-integration/actions/workflows/build-image.yaml/badge.svg)
![test badge](https://github.com/mozilla/jira-bugzilla-integration/actions/workflows/test-build.yaml/badge.svg)

# Jira Bugzilla Integration (JBI)
System to sync Bugzilla bugs to Jira issues.

### Caveats
- The system accepts webhook events from Bugzilla
- The events are transformed into Jira issues

> - The system POSTs an update to the see_also field of Bugzilla bugs

Author: ```suggestion
- The system sets the `see_also` field of the Bugzilla bug with the URL to the Jira issue
```

> - Bugs' whiteboard tags are used to determine if they should be synchronized or ignored

Reviewer: I would move this line in the second position

> - The system accepts webhook events from Bugzilla
> - The events are transformed into Jira issues
> - The system sets the `see_also` field of the Bugzilla bug with the URL to the Jira issue
> - Bugs' whiteboard tags are used to determine if they should be synchronized or ignored

Author: ```suggestion
- The system accepts webhook events from Bugzilla
- Bugs' whiteboard tags are used to determine if they should be synchronized or ignored
- The events are transformed into Jira issues
- The system sets the `see_also` field of the Bugzilla bug with the URL to the Jira issue
```

## Action Configuration

> The action configuration uses a YAML file per environment, this file stores all the configuration information for jbi data flows. The process to enable a new data flow will be explained below.

Reviewer: ```suggestion
The system reads the action configuration from a YAML file, one per environment. Each entry controls the synchronization between Bugzilla tickets with Jira issues.
```

> Below is an full example of an action configuration:

Reviewer: ```suggestion
Below is a full example of an action configuration:
```

```yaml
    action: src.jbi.whiteboard_actions.default
    contact: [example@allizom.com]
    description: example configuration
    enabled: true
    parameters:
      jira_project_key: EXMPL
      whiteboard_tag: example
```

A bit more about the different fields...
- `action` (optional)
    - string
    - default: [src.jbi.whiteboard_actions.default](src/jbi/whiteboard_actions/default.py)

>     - If using a custom action, place the PYTHONPATH to the custom action python module

Reviewer: ```suggestion
    - The specified Python module must be available in the `PYTHONPATH`
```

- `contact`
    - list of strings

>     - If an issue arises with the workflow, communication will be established with this `contact`

Reviewer: ```suggestion
    - If an issue arises with the workflow, communication will be established with these contacts
```

    - Please enter the contact information for one or more stakeholders
- `description`
    - string
    - Please enter a description; for example, team name or project use-case.
- `enabled` (optional)
    - bool [true, false]
    - default: false
    - If false, matching events will not be synchronized
- `parameters` (optional)
    - dict
    - default: {}

>     - The parameters will be validated to ensure the selected action has all expected parameters

Reviewer: ```suggestion
    - The parameters will be validated to ensure the selected action accepts the specified values
```
